src/components/App.tsx

```diff
@@ -502,16 +502,24 @@ class App extends React.Component<ExcalidrawProps, AppState> {
     history.clear();
   };
 
-  /** Completely resets scene & history.
-   * Do not use for clear scene user action. */
-  private resetScene = withBatchedUpdates(() => {
-    this.scene.replaceAllElements([]);
-    this.setState({
-      ...getDefaultAppState(),
-      appearance: this.state.appearance,
-    });
-    this.resetHistory();
-  });
+  /**
+   * Resets scene & history.
+   * ! Do not use to clear scene user action !
+   */
+  private resetScene = withBatchedUpdates(
+    (opts?: {
+      /** defaults to `true` */
+      resetIsLoading: boolean;
+    }) => {
+      this.scene.replaceAllElements([]);
+      this.setState((state) => ({
+        ...getDefaultAppState(),
+        isLoading: opts?.resetIsLoading !== false ? false : state.isLoading,
```

Comment:
can we rename `resetIsLoading` to `showLoader` instead which would be false/undefined by default to simplify ?
`isLoading: opts.showLoader ? false : state.isLoading`

Reply:
We can rename for sure, but I wouldn't change the logic. It's best to simply reuse whatever the current state is, otherwise we're asking for trouble down the line.

Reply:
Well, you're kinda doing that, too.

Reply:
So if you wanna invert the default, it should be `doNotResetLoader`. Also, the word `loader` is confusing/ambiguous, so it'd have to be `doNotResetLoadingState`.

Reply:
Maybe the default should be inverted, too. I don't imagine the `resetScene` cancelling I/O (and thus loading state). Thoughts?

Reply:
I think I got confused with it so we want to  have `resetIsLoading` always true by default which means `isLoading` will always be set to `false `by default unless host explicitly pass `resetIsLoading` as `false` as per the logic. So right now `undefined` and `true` are behaving same thats why I guess i got little confused.

>Maybe the default should be inverted, too. I don't imagine the resetScene cancelling I/O (and thus loading state). Thoughts?

Yes, I agree with this.

Reply:
> unless host explicitly pass `resetIsLoading` as `false` as per the logic. So right now `undefined` and `true` are behaving same thats why I guess i got little confused.

Yea, it may sound weird but it's perfectly reasonable pattern (provided it's what we want). In other words, this way it would not change current behavior except allow to override it.

Anyway, I've pushed the inversion.

Reply:
lgtm